sklearn/model_selection/tests/test_search.py

```diff
@@ -591,71 +552,212 @@ def test_param_sampler():
         assert_equal([x for x in sampler], [x for x in sampler])
 
 
-def test_randomized_search_grid_scores():
+def check_results_array_types(results, param_keys, score_keys):
+    # Check if the search results' array are of correct types
+    assert_true(all(isinstance(results[param], np.ma.MaskedArray)
+                    for param in param_keys))
+    assert_true(all(results[key].dtype is np.dtype('O')
+                    for key in param_keys))
+    assert_false(any(isinstance(results[key], np.ma.MaskedArray)
+                     for key in score_keys))
+    assert_true(all(results[key].dtype is np.dtype('float64')
+                    for key in score_keys if key != 'test_rank_score'))
+    assert_true(results['test_rank_score'].dtype is np.dtype('int'))
```

Comment:
One should never use python types such as int or float to build numpy dtypes as the precision level of the mapping of Python types to numpy dtypes can be platform dependent. Instead please be explicit and use `np.int32`, `np.int64`, `np.float32` or `np.float64` directly.

In particular this assertion is failing when run under Windows as reported by AppVeyor.